core/ClassInfo.php

```diff
@@ -52,17 +52,20 @@ public static function hasTable($class) {
 			return false;
 		}
 	}
-	
+
 	public static function reset_db_cache() {
 		self::$_cache_all_tables = null;
 		self::$_cache_ancestry = array();
 	}
-	
+
 	/**
 	 * Returns the manifest of all classes which are present in the database.
 	 * @param string $class Class name to check enum values for ClassName field
 	 */
 	public static function getValidSubClasses($class = 'SiteTree', $includeUnbacked = false) {
+		if (self::hasTable($class)) {
+			$class = self::$_cache_all_tables[strtolower($class)];
```

Comment:
The problem is that there's no guarantee that the class is ACTUALLY a sublcass of `DataObject`.

Also, I'm scared of the performance impact of creating singletons for every class just to get a normalised classname. strtolower is probably the least CPU intense way of making them consistent?